embrace-test-fakes/src/main/kotlin/io/embrace/android/embracesdk/fakes/FakeRequestExecutionService.kt

```diff
@@ -1,22 +1,29 @@
 package io.embrace.android.embracesdk.fakes
 
+import io.embrace.android.embracesdk.assertions.getSessionId
 import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
 import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
 import io.embrace.android.embracesdk.internal.delivery.execution.RequestExecutionService
 import io.embrace.android.embracesdk.internal.payload.Envelope
 import io.embrace.android.embracesdk.internal.payload.LogPayload
 import io.embrace.android.embracesdk.internal.payload.SessionPayload
+import io.embrace.android.embracesdk.internal.spans.findAttributeValue
+import io.opentelemetry.semconv.incubating.LogIncubatingAttributes
 import java.io.InputStream
 import java.util.concurrent.ConcurrentLinkedQueue
 import java.util.zip.GZIPInputStream
 
-class FakeRequestExecutionService : RequestExecutionService {
+class FakeRequestExecutionService(
+    private var strictMode: Boolean = true
+) : RequestExecutionService {
 
     private val serializer = TestPlatformSerializer()
     var constantResponse: ApiResponse = ApiResponse.Success(null, null)
     var responseAction: (intake: Envelope<*>) -> ApiResponse = { _ -> constantResponse }
     var exceptionOnExecution: Throwable? = null
     val attemptedHttpRequests = ConcurrentLinkedQueue<Envelope<*>>()
+    val sids = mutableSetOf<String>()
+    val lids = mutableSetOf<String>()
```

Comment:
nit: I think sessionIds and logIds would be more clear 😅

Reply:
Renamed variables